Add ASCII_DATE_TIME processing to special constants

jordanpadams opened the pull request:
## 🗒️ Summary
Added time conversions to Instant for comparison against constants.

## ⚙️ Test Data and/or Report
Automated unit tests below should all pass.

## ♻️ Related Issues
Closes #1135

Commit:
WIP

Commit:
add the test data

Commit:
fix up datetime processing

Needed an new problem type for this since they are not numbers as it is currently meant.

Used the DateTimeValidator to do the conversions from string to Instant. Turns out that the string needs some normalization and the pattern being used to convert also needs love depending on the value. Little more complicated than I would have liked.

Removed some dead code from FieldValueValidator because it would just make the error messages confusing and wrong.

Bulk of the update is in SpecialConstantChecker but is straight forward. The why is to support the ticket in general.

nutjob4life reviewed (commented):
Visual inspection looks good.

But the tests still fail … @al-niessner is aware of these issues.
```
[ERROR] Failures: 
[ERROR]   summary:totalErrors ==> expected: <2> but was: <3>
[ERROR]   summary:totalErrors ==> expected: <29> but was: <32>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalErrors ==> expected: <8> but was: <7>
[ERROR]   summary:totalWarnings ==> expected: <0> but was: <3>
[ERROR]   summary:totalWarnings ==> expected: <2> but was: <3>
[INFO] 
[ERROR] Tests run: 276, Failures: 16, Errors: 0, Skipped: 23
```

Commit:
update testing

It seems that more changes have occurred to context reference resulting in the changes to the feature file.

Have ot check if the BuffereedImage is null or not to know if it succesfully loaded an image. Added this and image checks are detetcting bad images.

Commit:
disable all of the pds.esa.int based tests
